src/commonMain/kotlin/fr/acinq/lightning/channel/Channel.kt

```diff
@@ -2335,82 +2342,107 @@ data class Negotiating(
-                            )
-                            Pair(nextState, actions)
-                        }
-                        nextClosingFee == event.message.feeSatoshis -> {
-                            // we have converged but they don't have our signature yet
-                            logger.info { "c:$channelId closing tx published: closingTxId=${signedClosingTx.tx.txid}" }
-                            val (_, closingSigned) = Helpers.Closing.makeClosingTx(keyManager, commitments, localShutdown.scriptPubKey.toByteArray(), remoteShutdown.scriptPubKey.toByteArray(), nextClosingFee)
-                            val nextState = Closing(
-                                staticParams,
-                                currentTip,
-                                currentOnChainFeerates,
-                                commitments,
-                                fundingTx = null,
-                                waitingSinceBlock = currentBlockHeight.toLong(),
-                                mutualCloseProposed = this.closingTxProposed.flatten().map { it.unsignedTx } + listOf(signedClosingTx),
-                                mutualClosePublished = listOf(signedClosingTx)
-                            )
-                            val actions = listOf(
-                                ChannelAction.Storage.StoreState(nextState),
-                                ChannelAction.Blockchain.PublishTx(signedClosingTx.tx),
-                                ChannelAction.Blockchain.SendWatch(WatchConfirmed(channelId, signedClosingTx.tx, staticParams.nodeParams.minDepthBlocks.toLong(), BITCOIN_TX_CONFIRMED(signedClosingTx.tx))),
-                                ChannelAction.Message.Send(closingSigned)
-                            )
-                            Pair(nextState, actions)
-                        }
-                        else -> {
-                            val (closingTx, closingSigned) = Helpers.Closing.makeClosingTx(keyManager, commitments, localShutdown.scriptPubKey.toByteArray(), remoteShutdown.scriptPubKey.toByteArray(), nextClosingFee)
-                            logger.info { "c:$channelId proposing closingFeeSatoshis=${closingSigned.feeSatoshis}" }
-                            val closingProposed1 = closingTxProposed.updated(
-                                closingTxProposed.lastIndex,
-                                closingTxProposed.last() + listOf(ClosingTxProposed(closingTx, closingSigned))
-                            )
-                            val nextState = this.copy(
-                                commitments = commitments.copy(remoteChannelData = event.message.channelData),
-                                closingTxProposed = closingProposed1,
-                                bestUnpublishedClosingTx = closingTx
```

Comment:
NB: there was a bug here: we must use `signedClosingTx` and not `closingTx`, otherwise we're forgetting our peer's signature :face_with_head_bandage: